src/net_processing.cpp

```diff
@@ -2472,8 +2471,11 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
             return;
         }
 
-        if (!vRecv.empty())
-            vRecv >> addrFrom >> nNonce;
+        if (!vRecv.empty()) {
+            // The pre-31402 serialization of addrFrom (which didn't include nTime).
+            vRecv >> Using<CustomUintFormatter<8>>(addrFrom.nServices) >> static_cast<CService&>(addrFrom);
```

Comment:
Since `addFrom` is just thrown away, maybe make this more explicit:

```diff
--- a/src/net_processing.cpp
+++ b/src/net_processing.cpp
@@ -2439,7 +2439,6 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
 
         int64_t nTime;
         CAddress addrMe;
-        CAddress addrFrom;
         uint64_t nNonce = 1;
         ServiceFlags nServices;
         int nVersion;
@@ -2472,8 +2471,12 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         }
 
         if (!vRecv.empty()) {
-            // The pre-31402 serialization of addrFrom (which didn't include nTime).
-            vRecv >> Using<CustomUintFormatter<8>>(addrFrom.nServices) >> static_cast<CService&>(addrFrom);
+            // The version message includes information about the sending node which we don't use:
+            //   - 8 bytes (service bits)
+            //   - 16 bytes (ipv6 address)
+            //   - 2 bytes (port)
+            uint8_t unused[26];
+            vRecv >> unused;
             vRecv >> nNonce;
         }
         if (!vRecv.empty()) {
```

Reply:
Agree that `addrFrom` can be removed completely if it is never read from

Reply:
Done (using `vRecv.Ignore(26);` instead).

Reply:
Thanks for teaching me about `CDataStream.ignore()`!